### PR TITLE
Hide the context menu on tap down

### LINKED ISSUE #125364: [Desktop] Selection toolbar buttons change before closing
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Run the code sample
2. Select a word in the text field and open selection toolbar
3. Tap elsewhere so the selection changes to collapsed

### Expected results

Selection toolbar options don't change until the toolbar is closed.

### Actual results

Available buttons change as selection is changed before the toolbar is closed.

selection-toolbar

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final TextEditingController _controller = TextEditingController(
    text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
  );

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text('Select a word and open the toolbar'),
            TextField(
              controller: _controller,
              style: const TextStyle(fontSize: 24.0),
            ),
          ],
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

_No response_

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
% flutter doctor -v
[✓] Flutter (Channel master, 3.10.0-12.0.pre.26, on macOS 13.3.1 22E261 darwin-arm64, locale en-ES)
    • Flutter version 3.10.0-12.0.pre.26 on channel master at /Users/tgucio/Library/Frameworks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d2646cf87d (12 hours ago), 2023-04-21 23:06:17 -0400
    • Engine revision b6749106a4
    • Dart version 3.1.0 (build 3.1.0-30.0.dev)
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/tgucio/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.77.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.3.1 22E261 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 112.0.5615.137

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by justinmc**
Previously, tapping to move the cursor while the context menu was open would update the contents of the text selection toolbar before closing it, so a briefly flash of change was seen.  This PR fixes it by correctly hiding the menu on tap down instead of up.

context_menu_down

Fixes https://github.com/flutter/flutter/issues/125364

### CODE REVIEW

File: packages/flutter/lib/src/widgets/text_selection.dart [hunk 1]
```diff
@@ -2108,10 +2108,13 @@ class TextSelectionGestureDetectorBuilder {
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
       case TargetPlatform.fuchsia:
+        // On mobile platforms the selection is set on tap up.
+        editableText.hideToolbar(false);
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/text_selection.dart:2112]: I tried this on Android with a physical mouse and the handle remains visible:

https://user-images.githubusercontent.com/389558/236953666-f3d8c797-6a5d-49ea-8825-d5cbc5a59512.mp4

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by justinmc** [comment]
The golden failure was not desired, I've pushed a commit to fix it.

**Comment by justinmc** [comment]
Actually I think the best move is to update the golden.  I tested on native Android 13 in the Keep app, and the handle remains visible after tapping outside of an existing selection (see the [existing test](https://github.com/flutter/flutter/blob/15a7e07395ed0701e3f2393d57d0159ef5a66529/packages/flutter/test/material/text_field_test.dart#L1109-L1112)).

**Pull request merged**